tests/test_realsense_d435.py

```diff
@@ -0,0 +1,14 @@
+from context import modules
+import unittest
+
+from modules import realsense_d435
+
+class TestTemplate(unittest.TestCase):
+    def setUp(self):
+        self.rs = realsense_d435.rs_d435()
+    
+    def tearDown(self):
+        pass
+
+    def test_import_success(self):
```

Comment:
Rob rasied a good point in that monkeywrenching could be useful here? You could override the pipe method to a class of your chosing and use that to feed data in? Then you could use that to ensure your private functions work as expected.

Reply:
Had a stab at some unit tests c6cd50b. Not convinced by all of them